### PR TITLE
[WIP] add Knex migrations

### PR DESCRIPTION
**Pull request by ashleygwilliams**
- [x] users table
- [x] projects table
- [x] pages table
- [x] elements table
- [ ] indexes
- [ ] update trigger
- [x] localization column transform
- [x] user_id on pages (temp)
- [x] user_id on elements (temp)
- [ ] backfill user_id on pages
- [ ] backfill user_id on elements

### CODE REVIEW

**Comment by ashleygwilliams** [comment]
@jbuck @cadecairos if you want to test this out rn:
```
$ npm install
$ knex migrate:latest
```

you know it worked if you see:
```
Using environment: development
Batch 1 run: 2 migrations
/Users/ag_dubs/Projects/api.webmaker.org/migrations/0_add_user_id_to_pages.js
/Users/ag_dubs/Projects/api.webmaker.org/migrations/1_add_user_id_to_elements.js
```

**Comment by cadecairos** [comment]
we should document that knex should be installed globally, or add an npm script to use the binary in node_modules

**Comment by ashleygwilliams** [comment]
+1 cade, documentation will def be included in this PR

**Comment by ashleygwilliams** [comment]
i think having an npm script that points to the node_modules version of knex is best practices, but i'm open to other ideas

**Comment by jbuck** [comment]
"use an npm script"++

File: knexfile.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      tableName: 'migrations'
+    }
+  },
+  testing: {
+    client: 'pg',
+    debug: true,
+    connection: 'postgresql://localhost:5432/webmaker-testing',
+    directory: './migrations',
+    migrations: {
+      tableName: 'migraitons'
```
Comment (jbuck) [inline knexfile.js]: i and t are swapped here

**Comment by jbuck** [comment]
@ashleygwilliams Could we switch to the "basic" knexfile syntax? Like:

```
// Load config into process.env
require('habitat').load();

module.exports = {
  client: 'pg',
  connection: process.env.POSTGRE_CONNECTION_STRING
};
```

Because it looks like our environment configuration is the same across all of our environments

**Comment by ashleygwilliams** [comment]
filed an issue https://github.com/tgriesser/knex/issues/924

File: package.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
   "main": "server.js",
   "scripts": {
     "start": "node server.js",
-    "migrate": "node scripts/migrate",
+    "migrate": "knex migrate:latest",
     "test": "npm run test:api && npm run lint",
     "test:api": "npm run test:setup && lab -t 100 -c --verbose --colors --assert code --timeout 5000",
     "test:setup": "npm run test:droptables && npm run test:createtables && npm run migrate -- 'tests.env' && npm run test:insertdata",
```
Comment (jbuck) [inline package.json]: need to remove npm run test:createtables here

**Comment by ashleygwilliams** [comment]
i have a uniqueness constraint on users.username and users.id yet these tests fail? @cadecairos are these supposed to throw on route validation? if so- this PR doesn't change any of that... so i'm not sure what's happening?

```
307) User Handlers Create Does not allow duplicate usernames:
      actual expected
      500400
      Expected 500 to equal specified value
      at /home/travis/build/mozilla/api.webmaker.org/test/services/api/handlers/users.js:52:36
  308) User Handlers Create Does not allow duplicate ids:
      actual expected
      500400
      Expected 500 to equal specified value
      at /home/travis/build/mozilla/api.webmaker.org/test/services/api/handlers/users.js:63:36
  319) User Handlers Patch Does not allow duplicate usernames:
      actual expected
      500400
      Expected 500 to equal specified value
      at /home/travis/build/mozilla/api.webmaker.org/test/services/api/handlers/users.js:210:36
```

**Comment by jbuck** [comment]
I think the server code is looking at the name of the index returned in the postgres error, and if knex changed the name of the unique index on username or user id, I could see this returning a generic server 500 error instead of having it be re-interpreted as a 400 error.

**Comment by ashleygwilliams** [comment]
ah, got it @jbuck. i can give it a name, if i make it an index- but again- did we really want an index on that column?

**Comment by jbuck** [comment]
Yes, we'll need those indices to enforce uniqueness constraints: http://www.postgresql.org/docs/9.4/static/indexes-unique.html

**Pull request closed without merging**